src/main/java/io/github/thebusybiscuit/slimefun4/api/blocks/EmptyBlockData.java

```diff
@@ -12,7 +12,9 @@
  * @author TheBusyBiscuit
  *
  */
-class EmptyBlockData extends SlimefunBlockData {
+final class EmptyBlockData extends SlimefunBlockData {
+
+    static EmptyBlockData INSTANCE = new EmptyBlockData();
```

Comment:
Having this static singleton here is unnecessary in my opinion.
We don't yet know whether we will even need this class.
And if we do, then the singleton should be a private field in the BlockDatabase rather than a static field to avoid unnecessary memory leaks when disabling the plugin.